handbook/engineering/search/index.md

```diff
@@ -2,9 +2,93 @@
+The answer to “Can I do ‘X’ with Sourcegraph?” will always be yes.
+- **Easy to use:** Users can figure out how to find what they are looking for, the syntax is clear, new users understand what options they have for searching.
+- **Sharable:** Sourcegraph search creates network effects that provide compounding value with scale. 
+
 ## Goals
 
-Our current goals are documented in the [search team's current tracking issue](https://github.com/sourcegraph/sourcegraph/issues?q=is%3Aopen+is%3Aissue+label%3Atracking+label%3Ateam%2Fsearch).
+_Updated 2020-08-27_
+
+## Long term (~1 year)
+
+Sourcegraph search is far and away the best code search solution, search results are fast, and I can find what I’m looking for quickly.
+
+## Medium term (3-6 months)
+
+In the next 6 months we want to be halfway to the long term goal. To achieve this, we need to accomplish the following outcomes:
+
+- Getting to search results faster through a better search experience.
+- What do we need to do to get to the speed of grep.app?
+- Scale fast search to 500k repositories.
+
+## Short term (1-3 months)
+
+Our current focus is documented in the [tracking issue for the current milestone](https://github.com/sourcegraph/sourcegraph/issues?q=is%3Aopen+is%3Aissue+label%3Atracking+label%3Ateam%2Fsearch).
+
+### Fast
+
+- **Streaming search**
+  - **Problem:** We want to make significant improvements to search that are blocked by not having the infrastructure needed.
+  - **Owner:** Keegan
+  - **Status:** In progress
+
+### Expressive
+
+- **Hierarchical search (AND/OR/NOT)**
+  - **Problem:** We have customers who want to migrate from OpenGrok and other search tools, and they want to be able to do searches that are available in those tools.
+  - **Outcomes:** Users can run complex searches on Sourcegraph.
+  - **Owner:** Rijnard
+  - **Status:** In progress
+- **Improve syntax for existing filters like `repohasfile`**
+  - **Problem:** Creating a custom filter name for every permutation of search use does not scale (existing filter examples: `repohasfile`, `hascommitafter`).
+  - **Outcomes:** We have an expressive syntax that scales and effectively leverages existing filters to achieve searches that previously would have required a custom/new filter.
+  - **Owner:** Rijnard/Stefan
+  - **Status:** Not started
+- **Revision search**
+  - **Problem:** Users want to search across branches with the same ease as searching across repositories, and are currently unable to do so.
+  - **Owner:** Stefan
+  - **Status:** In progress
+
+### Easy to use
+
+- **Get to files and repositories more quickly**
+  - **Problem:** Finding files and repositories often takes multiple tries to get the syntax correct or creative solutions to get the result users are looking for.
+  - **Outcomes:** Users leverage file and repo filters to quickly get to the code they are searching for.
+  - **Owner:** Stefan
+  - **Status:** In progress
```

Comment:
Does this refer to the wildcard syntax I am working on? If yes, I am not sure that the wildcard syntax addresses the problem statement. The assumption would be that users fail on regex and not on the filter syntax. Is that really the case? If we just want to increase the adoption of repo and file filters then maybe this is more of a UX topic?

Reply:
Yep, this is the higher level problem statement for your work on wildcard syntax. I’m curious if you think there’s a different problem statement the wildcard syntax solves that isn’t captured?
> The assumption would be that users fail on regex and not on the filter syntax.
> We just want to increase the adoption of repo and file filters then maybe this is more of a UX topic.

Strongly agree with both of your points - 1/ validating the assumption that users fail on regex and not the filter syntax through research and 2/ understanding the UX aspect as a part of the solution space for this problem. If we go the route of understanding the UX, we can add that as sub-bullet under _Plan_ (see latest commit for change).

Reply:
I agree it is more of a UX research topic right now. I suggest to remove the point `**Get to files and repositories more quickly**` entirely. Like I said above it is not clear to me that Wildcard syntax is related to this goal. I would suggest to first implement what is listed under `### Easy to use` and see whether adoption of our syntax goes up. 

Let's validate whether regex is indeed a widespread problem for our key target group before adding wildcard syntax to our short term goals.

Especially in the short term, I want to focus on the expressiveness of our query language first.